### DISCUSSION TITLE
Provide a way to disable emails for deployment reviews per repository/team

### DISCUSSION THREAD

**Discussion by diastremskii** [community/community#14564] [Enterprise]
Hello!

When using the deployment approval flow, by default, all reviewers will receive an email asking to review a deployment.
Example configuration enabling reviews:
<img width="783" alt="image" src="https://user-images.githubusercontent.com/20017097/162448804-856cdc9a-f4e6-4050-9e0b-0074f8597132.png">
When working on a project in a team where everyone can deploy, there is a lot of unnecessary emails being sent because of that. There is also a clear ownership of changes, the same person:
1. Opens a PR
2. Gets review and approval
3. Merges the changes
4. Tests in development
5. If everything is fine, approves the production deployment

I can understand that emails work for projects where there is a lot of contributors who can’t deploy and a small number of repo maintainers who are able to deploy to production. But for private or GHE projects it's just unnecessary to notify everyone who is able to approve deployments and it generates some annoyance.

Thanks!

**Comment by bramvonkmanucharsso** [comment]
We have another workflow, based on https://devblogs.microsoft.com/devops/i-need-manual-approvers-for-github-actions-and-i-got-them-now/

Every merge to main gets deployed to our development environment.

After that, we might want to push that same commit/PR merge to higher environments. As there are no other ways of manual approval/steps/buttons in github actions, we're using GitHub Environments with approvals to make that possible. 

But we only sometimes do that, not with every PR merge. But we get a lot of emails this way.

Therefore, we would like the same as @diastremskii : we'd love the possibility to disable specifically this type of email.

**Comment by c-wygoda** [comment]
Same here, the manual reviewer workflow for deployments is better UX than manual workflow starts, but we expect the PR owner to make sure everything get's deployed all the way. No reason to send an email every time an manual deployment review is needed as PR owner is expected to make sure someone is approving.

**Comment by DennisBecker** [comment]
My inbox explodes with those review emails - I want to choose from which teams I would get those review emails or just completely disable those emails.

**Comment by Dawnflash** [comment]
Same here, we have a workflow that needs manual reviewing but the reviewer pool is huge and we don't want to spam half the company. Please let us turn the notifications off or just notify the author.

**Comment by idexxbernvaug** [comment]
My team would love to be able to allow people to review but not send them an email - specifically for deployments to our dev environment.

**Comment by billhammond-dev** [comment]
This is also something we would like. In an org where code changes are 'open' to larger groups of people this feature cannot be used without spamming everybody. For now we have decided we cannot use the deployment approvals feature unless the repo in question is something that needs to be locked down to a smaller group.

**Comment by cbotsikas** [comment]
+1 one here.

**Comment by SergiiPo** [comment]
Please add silent option to not spam everybody.

**Comment by fxmarty** [comment]
+1

**Comment by jzamanxometry** [comment]
+1 please do this. It's really annoying to get notifications like this

**Comment by Ugzuzg** [comment]
> You are receiving this because you are subscribed to this thread.
> [Manage your GitHub Actions notifications](https://github.com/settings/notifications)

Additionally, the email contains the above footer suggesting that these notifications can be configured. In reality, there is no such option.

**Comment by tlgDJPorv** [comment]
+1

**Comment by evalexpr** [comment]
+1

**Comment by aoj2** [comment]
+1

**Comment by BrettZeidler** [comment]
This needs to be addressed, I'm getting bombarded with emails for deployment reviews in repositories I don't work in with no ability to disable these specific notifications without ignoring the entire repo in general (I can't even view these supposed review requests within the notifications inbox in Github itself), despite the email for these emails telling me there's an ability to customize or disable them in the notification settings.

**Comment by voiddev90** [comment]
+1 on this issue. We recently switched to GitHub.
We'd like to be able to turn off the deployment review notifications.

**Comment by alexdltr** [comment]
+1 please add this functionality

**Comment by schammah** [comment]
+1

**Comment by kotapeter** [comment]
We also need this functionality. For the time being, I've disabled the action notifications [here](https://github.com/settings/notifications).

<img width="524" alt="Screenshot 2024-01-19 at 13 19 36" src="https://github.com/community/community/assets/3911387/9ca2ab32-2b7f-426e-90e2-50ead02a7b7c">

**Reply by mtvx-o** [reply]
That disables _all_ GHA notifications, though.

**Comment by dtitov** [comment]
Any update? Still no remedy for spamming about deployment reviews...

**Comment by shobhit-tiwari-maersk** [comment]
+1 this is absolutely needed. Too much spam for deployment reviews

**Comment by ceferrari** [comment]
+1

**Comment by aygalinc** [comment]
+1 it is unmanageable in GHE

**Comment by iv0rish** [comment]
+1

**Comment by christianplatta1012** [comment]
+1, we have a dashboard that shows waiting pipelines. The additional email is only spam in our workflow

**Comment by nthornton2010** [comment]
+1

**Comment by Shereef** [comment]
+1

**Comment by acLebert** [comment]
+1

**Comment by cxyblood21** [comment]
+1, this creates a lot of spam in my email inbox with no way to pick and choose notification options

**Comment by uzzz** [comment]
+1

**Comment by francisconroy** [comment]
Yep, most times those "deployment review requested" emails are total waste as the person who just merged will run the deploy also... Please give us a way to turn it off.

**Comment by AnaisCR** [comment]
+1

**Comment by LP6546** [comment]
Hello,

Ce problème de spamming est connu par l’équipe GitHub Action : https://github.com/orgs/community/discussions/14564

Je vous conseille de créer un autre groupe GitHub ou changer le champs « reviewer » dans le workflow dependabot :
https://github.tools.digital.engie.com/GBSEngieDigitalNemoOpti/CubicwebNemo/blob/0ac35c93fa9f8713c4594b327ab706076a60b410/.github/dependabot.yml#L16

De : Anais ***@***.***>
Envoyé : mercredi 22 mai 2024 11:24
À : community/community ***@***.***>
Cc : BOUVRY Antonin (ENGIE IT) ***@***.***>; Manual ***@***.***>
Objet : Re: [community/community] Provide a way to disable emails for deployment reviews per repository/team (Discussion #14564)

+1 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread. Message ID: <community/community/repo-discussions/14564/comments/9518988@ github. com> ‍ ‍ ‍ ‍ ‍ ‍ ‍


+1

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/orgs/community/discussions/14564*discussioncomment-9518988__;Iw!!La4veWw!xnPNr63eNvnuYB917vIdHXnDUbm6k5Oo07q0hyRAiEtXVLBz1pqSIVmxrz1TsBf5-DDC7CMnpBdq5u2EZU08wTu-$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/BGAR7VM6FRXTPN247G45SZDZDRP2DAVCNFSM5S4UMQR2U5DIOJSWCZC7NNSXTOKENFZWG5LTONUW63SDN5WW2ZLOOQ5TSNJRHA4TQOA__;!!La4veWw!xnPNr63eNvnuYB917vIdHXnDUbm6k5Oo07q0hyRAiEtXVLBz1pqSIVmxrz1TsBf5-DDC7CMnpBdq5u2EZchrGzuV$>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.******@***.***>>

ENGIE Mail Disclaimer: http://www.engie.com/disclaimer/

**Comment by asmith-rh** [comment]
+1

**Comment by captainsamyak** [comment]
Same problem here, is this still not resolved? It's 2024 already!

**Comment by ns-dfarino** [comment]
+1

**Comment by ThomasHeinkel** [comment]
+1 Is there any workaround?

**Comment by ashic** [comment]
It's ridiculous that it's not implemented yet. Been nearly 2.5 years. +1 from me.

**Comment by nebuk89** [comment]
What would everyones preference be, would just having a nested tick box below required reviewers which was 'but don't send them emails' be enough for y'all here?

**Reply by tlgDJPorv** [reply]
The notifications have a link "Manage your GitHub Actions notifications", so intuitively, 
I was thinking https://github.com/settings/notifications under "System" | "Actions" | "Notify me", then like a tick box for "Deployment reviews" that could be unticked.

**Reply by ashic** [reply]
> What would everyones preference be, would just having a nested tick box below required reviewers which was 'but don't send them emails' be enough for y'all here?

I can think of two places (not mutually exclusive):

Since the review is on the environment, have an option there. i.e. for this env, notify only these reviewers (in addition to the current "required reviewers" option).
Allow a setting in the build.yml file that enables / disables notifications for particular steps (for all required reviewers).
While we can manage notifications via account settings now, it's a few clicks away and all or nothing.

**Reply by nthornton2010** [reply]
I like @tlgDJPorv and @ashic's ideas. Having a single setting under notifications, kind of as a master control for all and also an option on each environment and/or workflow. Personally, I don't want any notifications right now and would turn the master setting off for myself, but others on my team may want them on specific environments or workflows.

**Reply by nebuk89** [reply]
Amazing, thank you all for the engagement and feedback. I cannot say 'we are going to do this right now' but I will go and look to see if any of the 2-3 places y'all have suggested is something we can get done later this year.

**Reply by leovolving** [reply]
@nebuk89 Will this thread be updated when an update is available, or should be look elsewhere for updates?

**Reply by nebuk89** [reply]
I will make sure I come back to this thread :) All the PMs across Actions are going to make an effort to be more engaged in our community going forward and to ensure we are listening to you all/talking with you all here. <3

**Reply by dannybarnes** [reply]
Any ETA on when this feature will become available?

**Reply by OscarVanL** [reply]
Do you have any updates on this?

**Comment by ontramatt** [comment]
Any updates on this?

**Comment by glisibor** [comment]
Any update on this

**Comment by JonasRingeis** [comment]
Any updates? I really get spammed by these emails.

**Comment by ifrost** [comment]
+1

**Comment by Logan08691** [comment]
+1

**Comment by WolfgangRamos** [comment]
+1

**Comment by matlin-antura** [comment]
+1

**Comment by cra-schleupen** [comment]
+1

**Comment by fuzzypawzz** [comment]
+1

**Comment by oyedesam** [comment]
+1

**Comment by zenkaTheP** [comment]
+1

**Comment by zeeshan-sid** [comment]
+1

**Comment by iwinter-roo** [comment]
+1 this is a major problem with this functionality

**Comment by Keseg** [comment]
+1 I've had to disable reviews for our environments due to this reason.

**Comment by OutdatedVersion** [comment]
Hi all! Just wanted to share [a userscript](https://github.com/OutdatedVersion/github-deployment-approval-dismisser/tree/main) I recently wrote which dismisses these "approval requested" notifications in GitHub's native notification experience in-case someone else finds it useful! This has been working well for me as a stop-gap to native GitHub support as I'm not using their email notification feature.

**Comment by samirergaibi** [comment]
+1

**Comment by momirjalili** [comment]
+1

**Comment by hoivikaj** [comment]
+1

**Comment by kartvep** [comment]
+1

**Comment by ardove** [comment]
+1

**Comment by conrad90909** [comment]
+1

**Comment by ganesh-ns** [comment]
+1

**Comment by DavidBenavidez** [comment]
+1

**Comment by ghost** [comment]
+1

**Comment by arturasjakubauskas0109** [comment]
+1

**Comment by nadavbuc-orchid** [comment]
+1

**Comment by nbrugger-tgm** [comment]
+1

**Comment by joachim-tislov** [comment]
+1

**Comment by matsawczuk** [comment]
+1

**Comment by mpdibac-tmr** [comment]
+1

**Comment by SzymonOzog** [comment]
+1

**Reply by chris3ware** [reply]
Please upvote the original comment instead of commenting '+1'. We all get an email on an issue about reducing the amount of emails we get when you use '+1'.

**Reply by terumimcintyre-freightways** [reply]
oh the irony

**Comment by casey-robertson-paypal** [comment]
Adding a data point from a mid-size org: our Infrastructure Team is one of three teams listed as required reviewers on a production environment. Only one of the three needs to approve, but all three teams get notified on every single run, so we absorb the full notification volume for approvals we're rarely the right people to action.

The ask isn't just an email toggle — the underlying issue is that "listed as a possible approver" and "should be paged" are treated as the same thing. A per-environment or per-team option to be an eligible approver *without* being notified would solve this without anyone losing the ability to approve.
